Readability.js

```diff
@@ -245,10 +260,24 @@ Readability.prototype = {
       if (curTitle.split(' ').length < 3)
         curTitle = origTitle.replace(/[^\|\-]*[\|\-](.*)/gi,'$1');
     } else if (curTitle.indexOf(': ') !== -1) {
-      curTitle = origTitle.replace(/.*:(.*)/gi, '$1');
+      // XXX check if we have an heading containing this exact string, so we
+      // could assume it's the full title
+      // var headings = [].concat.call([].slice.call(doc.getElementsByTagName('h1')),
+      //                               [].slice.call(doc.getElementsByTagName('h2')));
+      var headings = this._concatNodeLists(
+        doc.getElementsByTagName('h1'),
+        doc.getElementsByTagName('h2')
+      );
+      var match = this._someNode(headings, function(heading) {
+        return heading.textContent === curTitle;
+      });
 
-      if (curTitle.split(' ').length < 3)
-        curTitle = origTitle.replace(/[^:]*[:](.*)/gi,'$1');
+      if (!match) {
+        curTitle = origTitle.replace(/.*:(.*)/gi, '$1');
+
+        if (curTitle.split(' ').length < 3)
+          curTitle = origTitle.replace(/[^:]*[:](.*)/gi,'$1');
```

Comment:
We added no supplementary replace; previous code was:

```js
    } else if (curTitle.indexOf(': ') !== -1) {
      curTitle = origTitle.replace(/.*:(.*)/gi, '$1');

      if (curTitle.split(' ').length < 3)
        curTitle = origTitle.replace(/[^:]*[:](.*)/gi,'$1');
    } else if (curTitle.length > 150 || curTitle.length < 15) {
```

New code is:

```js
    } else if (curTitle.indexOf(': ') !== -1) {
      // Wheck if we have an heading containing this exact string, so we
      // could assume it's the full title.
      var headings = this._concatNodeLists(
        doc.getElementsByTagName('h1'),
        doc.getElementsByTagName('h2')
      );
      var match = this._someNode(headings, function(heading) {
        return heading.textContent === curTitle;
      });

      // If we don't, let's extract the title out of the original title string.
      if (!match) {
        curTitle = origTitle.replace(/.*:(.*)/gi, '$1');

        if (curTitle.split(' ').length < 3)
          curTitle = origTitle.replace(/[^:]*[:](.*)/gi,'$1');
      }
    } else if (curTitle.length > 150 || curTitle.length < 15) {
```

The previous code colon-test codeblock has been simply moved into the new `if (!match)` one.